### ISSUE TITLE
Missing JSDoc param in .vue file

### ISSUE THREAD

**Issue by Holiden** [gajus/eslint-plugin-jsdoc#719]
Hello!

## Expected behavior

It is expected that the plugin will not display a missing parameter error as they are already defined.

## Actual behavior

I am getting an error about missing parameters. However, my JSDoc are filled in correctly. This is .vue file.

![1](https://user-images.githubusercontent.com/48956742/116499138-9b2f4980-a8d5-11eb-832f-5607f644fbaa.jpg)
![2](https://user-images.githubusercontent.com/48956742/116499147-9d91a380-a8d5-11eb-90cd-dacde1b93029.jpg)
![3](https://user-images.githubusercontent.com/48956742/116499152-9f5b6700-a8d5-11eb-8f36-7e34fedca296.jpg)

If use the suggested autofix, then the parameters are inserted before existing JSDoc declaration

![4](https://user-images.githubusercontent.com/48956742/116499347-2c062500-a8d6-11eb-89fd-570e1531bc15.jpg)

When editing a file anywhere, errors disappear. Not always, but sometimes it works

## ESLint Config

```js
module.exports = {
  plugins: ['jsdoc', 'vue'],
  parser: 'vue-eslint-parser',
  parserOptions: {
    parser: '@babel/eslint-parser',
    sourceType: 'module',
  },
  rules: {
    'jsdoc/check-alignment': 'error',
    'jsdoc/check-param-names': ['error', { checkRestProperty: true }],
    'jsdoc/check-property-names': 'error',
    'jsdoc/check-tag-names': 'error',
    'jsdoc/check-types': 'error',
    'jsdoc/newline-after-description': 'error',
    'jsdoc/no-bad-blocks': 'error',
    'jsdoc/no-undefined-types': 'error',
    'jsdoc/require-description': 'error',
    'jsdoc/require-description-complete-sentence': 'error',
    'jsdoc/require-hyphen-before-param-description': 'error',
    'jsdoc/require-param': ['error', { checkRestProperty: true, enableFixer: false }],
    'jsdoc/require-param-description': 'error',
    'jsdoc/require-param-name': 'error',
    'jsdoc/require-param-type': 'error',
    'jsdoc/require-property': 'error',
    'jsdoc/require-property-description': 'error',
    'jsdoc/require-property-name': 'error',
    'jsdoc/require-property-type': 'error',
    'jsdoc/require-returns': 'error',
    'jsdoc/require-returns-check': 'error',
    'jsdoc/require-returns-description': 'error',
    'jsdoc/require-returns-type': 'error',
    'jsdoc/require-throws': 'error',
    'jsdoc/require-yields': 'error',
    'jsdoc/require-yields-check': 'error',
    'jsdoc/valid-types': 'error',
  }
}
```

## Environment

- Node version: v14.16.1
- ESLint version: v7.24.0
- `eslint-plugin-jsdoc` version: v32.3.3
- IDE: Php Storm 2021.1.1

**Comment by brettz9** [comment]
If you can create a minimal test case repo, there is a greater chance we can debug this. I don't know it if may relate to your Webpack or Rollup routine.

**Comment by Holiden** [comment]
> If you can create a minimal test case repo, there is a greater chance we can debug this. I don't know it if may relate to your Webpack or Rollup routine.

Yes of course. [Here it is](https://github.com/Holiden/forJSDoc).
src/App.vue - all the same.

**Comment by brettz9** [comment]
Using the command line, `$(npm bin)/eslint --ext=vue .`, I only get one unrelated error, "Missing an explicit type attribute for button  vue/html-button-has-type".

When viewing within Atom, I get an error about no Babel config file being detected, but when I add `babelOptions` like thus (and I wasn't able to use `~/` for some reason):

```js
  parserOptions: {
    babelOptions: {
      cwd: '/Users/brett',
      configFile: './forJSDoc/babel.config.js'
    },
    parser: '@babel/eslint-parser',
    sourceType: 'module',
  },
```

...I also just get the same error as on the command line.

Perhaps this is an issue specific to your IDE.

**Comment by Holiden** [comment]
> Perhaps this is an issue specific to your IDE.

I tried opening it in VSCode and got the same errors

![image](https://user-images.githubusercontent.com/48956742/116523894-f165b200-a900-11eb-9db2-6a5e58791a88.png)

**Comment by Holiden** [comment]
I will clarify what it is in the files `.vue`. In files `.js` everything is correct.

**Comment by brettz9** [comment]
Do you have Atom available to test? I'm not sure why it would be working there... What was the original IDE you were using?

**Comment by Holiden** [comment]
> Do you have Atom available to test? I'm not sure why it would be working there... What was the original IDE you were using?

> * IDE: Php Storm 2021.1.1

I do not have Atom, but i can set it for test.

**Comment by brettz9** [comment]
I'm afraid I'm rather tied up in other projects to really explore this too deeply atm myself.

Hopefully someone can go to `/node_modules/eslint-plugin-jsdoc/dist/rules/requireParam.js` and add logging (or throwing errors as need be for the IDE to pick it up) around the `Missing JSDoc` to find out where things are going wrong.

Or perhaps it can be solved by adjusting the IDE's settings, e.g., ala https://github.com/babel/babel-eslint/issues/769#issuecomment-511218150 , in case your `@vue/cli-plugin-babel/preset` is required but is not getting discovered.

Hope that leads to some avenues of exploration, as I'm not really too sure at this point.

**Comment by gcunsolo** [comment]
Hi all,

I'm having the very same issue when working on a **Windows 10 OS**.

**Please note that I got no problem at all when using MAC OS.**

Here's a sample of a file I'm trying to lint:

```js
/**
     * @description My beautiful description
     * @param {string} param - This is an awesome param indeed
     * @return {boolean} true if not false
     */
    myBeautifulMethod(param) { ... }
```

Here's the output of my lint job:
```
   97:5  error  Missing JSDoc @description declaration        jsdoc/require-description
   97:5  error  Missing JSDoc @param "param" declaration     jsdoc/require-param
   97:5  error  Missing JSDoc @return declaration             jsdoc/require-returns
```

It's like the plugin found a JSDoc block, but was totally unable to analyze its content.

Just like @Holiden, If I try an `autofix` I got this result:
```
> 59 |     * @param method
     |       ^
  60 | /**
  61 |      * @description My beautiful description
  62 |      * @param {string} param - This is an awesome param indeed
```

From where I stand, it looks like a bad detection of end-of-line after the `/**`

**Comment by gcunsolo** [comment]
> When editing a file anywhere, errors disappear. Not always, but sometimes it works

Again, I got the very same behaviour.

@Holiden Can you please confirm my lead about Windows OS? Thanks in advance

**Comment by brettz9** [comment]
@gcunsolo : If you can use `plugins` with `jsdoc` and add the rules individually (instead of using `extends` with `plugins:jsdoc/recommended), can you try removing one half of the rules and through the process of elimination, identify which single rule may be sufficient to trigger the problem (is it `require-param` alone?)

Then, if you could reduce your code down to see whether that comment block alone in a file is sufficient to trigger the error...

And if you would also try upgrading to the latest `eslint-plugin-jsdoc`, since the reported version is several versions earlier now, and it is possible there was some fix.

**Comment by gcunsolo** [comment]
Thank you @brettz9 

I'm currently using 33.1.1, I'll give 34.0.2 a shot and get back to you with more information, as you asked!

(...and hey! Good job! We all get here to complain when there's something wrong... we never take the time to thank you all for your work!!)

**Comment by Holiden** [comment]
> @Holiden Не могли бы вы подтвердить мое мнение об ОС Windows? заранее спасибо

All right. This is the problem I have on OS Windows 10.

**Comment by brettz9** [comment]
While this doesn't look necessarily Windows specific, you might perhaps take a look at this JSDoc stripping issue referenced at https://github.com/jsdoc2md/jsdoc-to-markdown/wiki/How-to-document-TypeScript#tips--tricks

**Comment by gcunsolo** [comment]
Hi @brettz9 
Here's the result of my latest tests (v 34.8.2)

The "broken" rules are `require-description`, `require-param` and `require-returns`.
Executing `lint` with these rule gives unexpected results (see [previous comment](https://github.com/gajus/eslint-plugin-jsdoc/issues/719#issuecomment-839806170))
 
I confirm that `require-param` alone triggers the files corruption problem on autofix.
(expected, since it's the only rule having autofix capabilities)

For your information, I executed your unit tests and I got the following output:

```                                                                                                                                 
Rule, `require-param`, missing line numbers in errors: 24                                                                                                                                                                                                                      
  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬]                                                                                                                                                                                                                                       
  1943 passing (6s)                                                                                                                                                                                                                                                            
```

Tests are passing, and yet that log concerning `require-param` makes me curious :) 

https://github.com/gajus/eslint-plugin-jsdoc/blob/9fb6b36dc9fcb6362ebd9e6569d26fa68865b514/test/rules/index.js#L48-L55

As I said before, it looks like **only when using Windows** the plugin is able to identify a JSDoc block but totally unable to parse its content.

As an additional note, I tried to play with your `iterateJsdoc` to replace the `/n` at lines 139 and 159 with no luck

https://github.com/gajus/eslint-plugin-jsdoc/blob/6b6d90889163658e45f5e3c5c1dc4a0a0d50f2b7/src/iterateJsdoc.js#L114-L162

**Comment by brettz9** [comment]
Hi @gcunsolo ,

> Here's the result of my latest tests (v 34.8.2)
> 
> The "broken" rules are `require-description`, `require-param` and `require-returns`.
> Executing `lint` with these rule gives unexpected results (see [previous comment](https://github.com/gajus/eslint-plugin-jsdoc/issues/719#issuecomment-839806170))
> 
> I confirm that `require-param` alone triggers the files corruption problem on autofix.
> (expected, since it's the only rule having autofix capabilities)

Thanks for this info.
 
> For your information, I executed your unit tests and I got the following output:
> 
> ```
> Rule, `require-param`, missing line numbers in errors: 24                                                                                                                                                                                                                      
>   [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬]                                                                                                                                                                                                                                       
>   1943 passing (6s)                                                                                                                                                                                                                                                            
> ```
> 
> Tests are passing, and yet that log concerning `require-param` makes me curious :)

Yeah, I hard-coded this warning. We need to fix `require-param` to report more accurate line numbers. I didn't have time to complete my work on that review, so I put that warning to avoid adding bad test expectations which would satisfy the tests and avoid the need for that warning, but which would make it harder to find the offending line numbers when I or someone got around to fixing them. In short, this shouldn't be connected to our current problems with this issue.
 
> As I said before, it looks like **only when using Windows** the plugin is able to identify a JSDoc block but totally unable to parse its content.
> 
> As an additional note, I tried to play with your `iterateJsdoc` to replace the `/n` at lines 139 and 159 with no luck
> 
> https://github.com/gajus/eslint-plugin-jsdoc/blob/6b6d90889163658e45f5e3c5c1dc4a0a0d50f2b7/src/iterateJsdoc.js#L114-L162

Would need to take a closer look, but I would suspect Windows issues might be related to the recently discovered issue highlighted at https://github.com/gajus/eslint-plugin-jsdoc/issues/745#issuecomment-846424782 where the comment parser is dropping carriage returns. (I since reported the issue to https://github.com/syavorsky/comment-parser/issues/129 and am now awaiting a reply).

**Comment by RonnyTite** [comment]
Hi @brettz9 ,

the issue is still happening even with  syavorsky/comment-parser#129  closed.

So i have made a fork  to reproduce this issue on my Windows.  Basically i think its all related to line endings, windows files being CRLF .

fork can be found here : https://github.com/solven-eu/eslint-plugin-jsdoc/tree/jsdoc_throw_errors

here a litlle details , i've done to reproduce it:

- change the `.editorconfig` `end_of_line` property to `crlf` 
- add a folder named 'errorToReport` that contains my tested files ( 1 .vue _appToReport.vue_  and 1 .js _toReport.js_ )
- add a new command  in the package.json `lintReport` to lint fix only the .vue file ( because cannot reproduce on .js file)
- Install some plugin and made some changes on .eslintrc.json, to be compatible with .vue files

AND had to rename the name of the project to _eslint-plugin-jsdoc-fork_ to be able to install eslint-plugin-jsdoc.. inside the project 😅 

To reproduce , just launch the command `lintReport` and take a look at _appToReport.vue_

#### BEFORE

![image](https://user-images.githubusercontent.com/7472778/135620067-29d06cb7-7c26-4989-b54f-e0bc93cdb89a.png)

https://github.com/solven-eu/eslint-plugin-jsdoc/blob/dfc2ef8358c09eaa2ce093f979e0cff2372bc546/src/errorToReport/AppToReport.vue#L7-L17

#### AFTER

![image](https://user-images.githubusercontent.com/7472778/135611266-30079846-ceac-408c-937d-b596fc5d19e7.png)



and here the logs 

```
   9:5  error  Unexpected comment inline with code     no-inline-comments
   9:5  error  Missing JSDoc @description declaration  jsdoc/require-description
  12:0  error  Missing JSDoc @param "arg" description  jsdoc/require-param-description

```


Hopes this can help

**Comment by brettz9** [comment]
ESLint apparently consumes these (per https://eslint.org/docs/developer-guide/working-with-custom-parsers#all-nodes ), but it looks like it would come from a parser.

**Comment by brettz9** [comment]
I've confirmed that this is a bug, but a bug apparently with `vue-eslint-parser`. I've filed https://github.com/vuejs/vue-eslint-parser/issues/145 in the hopes they can handle it. Will close for now, as should be resolved once they fix this, though feel free to comment further upon definitive status updates there (e.g., if they may fix the issue and there are no longer problems or there are still problems). Thanks for the details, all, which helped better identify the source of the problem.

**Comment by brettz9** [comment]
Just reporting back that as per https://github.com/vuejs/vue-eslint-parser/issues/145 , this issue should now be fixed, and that fix was included in version 8.3.0 of `vue-eslint-parser`.

Please confirm if it is working for you all. Thanks!
